src/renderer.h

```diff
@@ -343,13 +404,71 @@ class Renderer {
     void setWireframeMode(bool flag = false);
 
     // Viewport
-    void setWorldView(struct matrix* matrix);
+    void setViewMatrix(struct matrix* matrix);
+    void setWorldViewMatrix(struct matrix* matrix);
     void setD3DViweport(struct matrix* matrix);
     void setD3DProjection(struct matrix* matrix);
 
+    // Lighting
+    void updateLightMatrices(struct boundingbox* sceneAabb);
+    void setWorldLightDir(float dirX, float dirY, float dirZ);
+    struct point3d getWorldLightDir();
+    void setLightIntensity(float intensity);
+    float getLightIntensity();
+    void setLightColor(float r, float g, float b);
+    point3d getLightColor();
+    void setAmbientIntensity(float intensity);
+    float getAmbientIntensity();
+    void setAmbientLightColor(float r, float g, float b);
+    point3d getAmbientLightColor();
+
+    // Material
+    void setRoughness(float roughness);
+    float getRoughness();
+    void setMetalness(float metalness);
+    float getMetalness();
+
+    // Shadow (common)
+    void setShadowFaceCullingEnabled(bool isEnabled);
+    bool isShadowFaceCullingEnabled();
+    void setShadowMapResolution(int size);
+    int getShadowMapResolution();
+    void setShadowConstantBias(float bias);
+    float getShadowConstantBias();
+
+    // Battle Shadow
+    void setShadowMapArea(float area);
+    float getShadowMapArea();
+    void setShadowMapNearFarSize(float size);
+    float getShadowMapNearFarSize();
+
+    // Field Shadow
+    void setFieldShadowMapArea(float area);
+    float getFieldShadowMapArea();
+    void setFieldShadowMapNearFarSize(float size);
+    float getFieldShadowMapNearFarSize();
+    void setFieldShadowOcclusion(float value);
+    float getFieldShadowOcclusion();
+    void setFieldShadowFadeStartDistance(float value);
+    float getFieldShadowFadeStartDistance();
+    void setFieldShadowFadeRange(float value);
+    float getFieldShadowFadeRange();
+    void setWalkmeshExtrudeSize(float size);
+    float getWalkmeshExtrudeSize();
+    void setWalkmeshPosOffset(float size);
+    float getWalkmeshPosOffset();
+
+    // Lighting Debug
+    void setHide2dEnabled(bool isEnabled);
+    bool isHide2dEnabled();
+    void setShowWalkmeshEnabled(bool isEnabled);
+    bool isShowWalkmeshEnabled();
+
     // Internal coord calculation
     uint16_t getInternalCoordX(uint16_t inX);
     uint16_t getInternalCoordY(uint16_t inY);
 };
 
 extern Renderer newRenderer;
+
+void lighting_debug(bool* isOpen);

```

Comment:
This one must be moved in a relative header file ( `light.h` maybe? )

Reply:
I moved it to the newly added `lighting_debug.h` file.